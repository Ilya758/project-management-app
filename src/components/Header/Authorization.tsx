import { NavLink } from 'react-router-dom';
import './Header.css';

export const Authorization = () => {
  const language = ['Ru', 'En'];
  return (
    <nav>
      <button className="language">{language[0]}</button>
      <NavLink to="/authentication/login" className="signUp">
        <i className="fas fa-user" aria-hidden="true"></i>
      </NavLink>
    </nav>
  );
};
