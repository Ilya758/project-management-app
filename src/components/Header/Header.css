header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  background: #262637;
  position: sticky;
  top: 0;
  z-index: 2;
  width: 100%;
  flex: 0 0 auto;
  height: 100px;
  box-shadow: 1px 8px 8px 0px rgba(34, 60, 80, 0.2);
  font-family: 'Open Sans', sans-serif;
  font-weight: 400;
  font-style: normal;
  padding: 10px;
}

header > a,
.linkMovie {
  color: #00deb6;
  text-decoration: none;
  padding: 0 1rem;
  font-size: 18px;
}

header > a:hover {
  text-decoration: underline;
}

nav {
  width: 80px;
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: center;
}

.container-nav-button {
  width: 100%;
  display: flex;
  justify-content: space-between;
}

.nav-container {
  width: 300px;
  height: 100%;
  display: flex;
  align-items: center;
}

.nav-container-link {
  width: 150px;
  height: 100%;
  display: flex;
  flex-direction: column;
  align-items: baseline;
  justify-content: center;
  margin: 0 3%;
}

.nav-container-link > a,
.delete-user {
  transition: 0.5s;
  color: #61dafb;
}

.nav-container-link > a:hover {
  color: #00deb6;
}
.delete-user:hover {
  color: orangered;
}

.logo {
  width: 150px;
  height: 90%;
  background: url('../../assets/jpg/project-manager-logo.jpg') no-repeat center center;
  background-size: cover;
  box-shadow: 4px 4px 8px 0 rgb(0 188 212 / 56%);
  cursor: pointer;
  margin: 0 1%;
}

.delete-user {
  border: none;
  background: no-repeat;
  color: #61dafb;
  font-family: -apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Oxygen, Ubuntu, Cantarell,
    Fira Sans, Droid Sans, Helvetica Neue, sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 16px;
  cursor: pointer;
}

.signUp,
nav > button {
  border: 1px solid #00deb6;
  border-radius: 3px;
  padding: 5px 10px;
  cursor: pointer !important;
  transition: 0.5s;
}

nav > button {
  background: none;
  color: #61dafb;
}
.signUp > i {
  margin-right: auto;
}

.signUp:hover,
nav > button:hover {
  border-color: #61dafb;
  color: #00deb6;
  box-shadow: 4px 4px 8px 0 rgb(0 188 212 / 56%);
}
