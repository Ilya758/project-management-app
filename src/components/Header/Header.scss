.header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  background: #262637;
  position: sticky;
  top: 0;
  z-index: 2;
  width: 100%;
  flex: 0 0 auto;
  height: 100px;
  box-shadow: 1px 8px 8px 0px rgba(34, 60, 80, 0.2);
  font-family: 'Open Sans', sans-serif;
  font-weight: 400;
  font-style: normal;
  padding: 10px;
}

.nav {
  width: 120px;
  height: 100%;
  display: flex;
  flex-wrap: wrap-reverse;
  align-items: center;
  justify-content: space-evenly;
}

.container-nav-button {
  width: 100%;
  display: flex;
  justify-content: space-between;
}

.nav-container {
  width: 300px;
  height: 100%;
  display: flex;
  align-items: center;
}

.container-link {
  width: 250px;
  height: 100%;
  display: flex;
  flex-direction: column;
  align-items: baseline;
  justify-content: space-around;
  margin: 0 3%;
}

.link-profile,
.delete-user {
  transition: 0.5s;
  color: #61dafb;
  text-decoration: none;
  border: none;
  background: no-repeat;
  cursor: pointer;
  font-size: 16px;
}

.link-profile:hover {
  color: #00deb6;
}
.delete-user:hover {
  color: orangered;
}

.logo {
  width: 150px;
  height: 90%;
  background: url('../../assets/jpg/project-manager-logo.jpg') no-repeat center center;
  background-size: cover;
  box-shadow: 4px 4px 8px 0 rgba(2, 189, 214, 0.56);
  cursor: pointer;
  margin: 0 1%;
}

.delete-user {
  border: none;
  background: no-repeat;
  color: #61dafb;
  font-weight: 400;
  font-style: normal;
  font-size: 16px;
  cursor: pointer;
}

.signUp,
.language,
.login-link,
.registration-link {
  border: 1px solid #00deb6;
  border-radius: 3px;
  padding: 5px;
  cursor: pointer !important;
  transition: 0.5s;
  display: block;
  width: 41px;
  text-align: center;
}

.language {
  background: none;
  color: #61dafb;
}

.signUp > .fas.fa-user,
.signUp > .fas.fa-user-plus {
  margin-right: auto;
  color: #61dafb;
}

.signUp > .fas.fa-user:hover,
.signUp > .fas.fa-user-plus:hover,
.language:hover {
  border-color: #61dafb;
  color: #00deb6;
  box-shadow: 4px 4px 8px 0 rgba(2, 189, 214, 0.56);
}
