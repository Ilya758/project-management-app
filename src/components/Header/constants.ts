export enum Classnames {
  headerOnScrollActive = 'header_onscroll_active',
}
