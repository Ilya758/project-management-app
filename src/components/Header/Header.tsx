import { NavLink } from 'react-router-dom';
import { Authorization } from './Authorization';
import './Header.scss';
import { SearchPage } from './Search/SearchPage';

export const Header = () => {
  return (
    <header className="header">
      <div className="nav-container">
        <NavLink to="/" className="logo"></NavLink>
        <SearchPage />
      </div>
      <Authorization />
    </header>
  );
};
