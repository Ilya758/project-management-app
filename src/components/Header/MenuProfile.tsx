import { NavLink } from 'react-router-dom';
import './Header.css';

export const Menu = () => {
  return (
    <div className="nav-container-link">
      <NavLink to="/board">Create new board</NavLink>
      <NavLink to="/edit_profile">Edit profile</NavLink>
      <NavLink to="/login">logout</NavLink>
      <button className="delete-user">Delete User</button>
    </div>
  );
};
