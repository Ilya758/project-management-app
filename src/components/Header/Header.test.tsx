import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Header } from './Header';

describe('Header Component', () => {
  it('Should render', () => {
    const component = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    );
    expect(component.baseElement).toMatchSnapshot();
  });
});
