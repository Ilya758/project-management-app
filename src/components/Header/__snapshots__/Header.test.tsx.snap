// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Should render 1`] = `
<body>
  <div>
    <header>
      <div
        class="nav-container"
      >
        <a
          aria-current="page"
          class="logo active"
          href="/"
        />
        <div
          class="nav-container-link"
        >
          <a
            class=""
            href="/board"
          >
            Create new board
          </a>
          <a
            class=""
            href="/edit_profile"
          >
            Edit profile
          </a>
          <a
            class=""
            href="/login"
          >
            logout
          </a>
          <button
            class="delete-user"
          >
            Delete User
          </button>
        </div>
      </div>
      <nav>
        <button
          class="language"
        >
          Ru / En
        </button>
        <button>
          Log In
        </button>
        <button>
          Sign Up
        </button>
      </nav>
    </header>
  </div>
</body>
`;
